### PR TITLE
unoptimised fibonacci

### PR DESCRIPTION
**Pull request by deosjr**

### CODE REVIEW

**Comment by deosjr** [comment]
A different usecase to experiment with concurrency on. So far, replacing the Conde in fib() with one using concurrent disj/conj slows everything down, but only replacing the inner ConjPlus with it's concurrent version speeds things up by roughly 10%

**Comment by deosjr** [comment]
Actually, interestingly, in this case _not_ wrapping each goal in Conj with Zzz is slower. That was what I was comparing to the sequential case before. If I wrap each subgoal in Zzz like in the paper, the concurrent Conj implementation speeds up this fibonacci implementation by 25% at least.

File: mini/fibonacci_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+// peano numbers and extralogical convenience functions
+var zero = ast.NewInt(0)
+var one = makenat(1)
+
+/*
+   (define (succ x y)
+       (== y `(,x)))
+*/
+func succ(prev, next *ast.SExpr) micro.Goal {
```
Comment (awalterschulze) [inline mini/fibonacci_test.go:18]: These nat stuff is nice enough to put into a `nat` folder.
Not in this pull request but in a following one, what do you think?

**Review by awalterschulze — APPROVED** [review]
But lets merge this first, this looks great.

File: mini/fibonacci_test.go (hunk 1 above)
Comment (deosjr) [reply mini/fibonacci_test.go:18]: If we put everything in its separate folder sure; if we put all the examples in one (which was what I had in mind before) then it doesnt really make sense to single these out.

Comment (awalterschulze) [reply mini/fibonacci_test.go:18]: I just think `nat` would be useful to a lot of use cases.

**Pull request merged**